scipy/ndimage/tests/test_filters.py

```diff
@@ -2014,7 +2014,39 @@ def test_rank15(self, dtype, xp):
                             [5, 6, 9, 3, 5]], dtype=dtype)
         output = ndimage.rank_filter(array, 0, footprint=footprint,
                                      origin=[-1, 0])
-        assert_array_almost_equal(expected, output)
+        xp_assert_equal(expected, output)
+
+    def test_rank16(self, xp):
+        array = [3, 2, 5, 1, 4]
+        expected = np.asarray([3, 3, 2, 4, 4])
```

Comment:
```suggestion
        array = xp.asarray([3, 2, 5, 1, 4])
        expected = xp.asarray([3, 3, 2, 4, 4])
```

Reply:
Thanks for your help. In this case, I rather avoid the modification. This test was created following @ev-br 's request from [here](https://github.com/scipy/scipy/pull/20543#pullrequestreview-2441175726):

> a test that rank_filter keeps working with input being a list

Which seems to be exactly what was requested.

Reply:
Indeed, I asked to add this test to make sure lists are still being accepted. Have now added a code comment to clarify.